lib/components/artworks/index.js

```diff
@@ -7,32 +7,51 @@ const { Text, View, StyleSheet } = React;
 import Separator from '../separator'
 import SerifText from '../text/serif';
 import ArtworksGrid from './grid';
+const { artworksQuery } = ArtworksGrid;
+
 import colors from '../../../data/colors';
 
 class Artworks extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { completedForSaleWorks: false };
+  }
+
   render() {
     const for_sale_count = this.props.artist.counts.for_sale_artworks;
     const other_count = this.props.artist.counts.artworks - for_sale_count;
     if (for_sale_count === 0) {
-      return this.renderSection('Works', other_count, this.props.artist.not_for_sale_artworks, true);
+      return this.renderSection('Works', other_count, this.props.artist.not_for_sale_artworks);
     } else {
+      const otherWorks = [];
+      const showOtherWorks = for_sale_count < 10 || this.state.completedForSaleWorks;
+      if (showOtherWorks) {
+        otherWorks.push(<Separator key='separator' />);
+        otherWorks.push(this.renderSection('Other Works',
+                        other_count,
+                        this.props.artist.not_for_sale_artworks,
+                        (artistID, page) => ArtworksGrid.artworksQuery(artistID, 'IS_NOT_FOR_SALE', page)));
+      }
       return (
         <View>
-          {this.renderSection('Works for Sale', for_sale_count, this.props.artist.for_sale_artworks, false)}
-          <Separator/>
-          {this.renderSection('Other Works', other_count, this.props.artist.not_for_sale_artworks, true)}
+          {this.renderSection('Works for Sale',
+                              for_sale_count,
+                              this.props.artist.for_sale_artworks,
+                              (artistID, page) => ArtworksGrid.artworksQuery(artistID, 'IS_FOR_SALE', page),
+                              () => { this.setState({ completedForSaleWorks: true }) })}
+          {otherWorks}
```

Comment:
This is where the React shadow DOM magic happens. Because we always render the ‘for sale’ section in the same place and only conditionally _add_ the ‘other works’ section, React knows that it needs to leave the existing (fully paginated) ‘for sale’ instance on screen _without_ us having to keep a reference to that section instance and have extra logic to create a new instance or re-use the existing one.